core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingRepositorySpec.scala

```diff
@@ -171,4 +172,29 @@ class BillingRepositorySpec extends AnyFlatSpec with TestDriverComponent {
       Await.result(repo.getLandingZoneId(billingProject.projectName), Duration.Inf)
     }
   }
+
+  behavior of "getBillingProjectsWithProfile"
+
+  it should "return billing projects with the specified profile ID" in withDefaultTestDatabase {
+    val repo = new BillingRepository(slickDataSource)
+    val firstProject = makeBillingProject()
+    val secondProject = makeBillingProject()
+    val billingProfileId = UUID.fromString(firstProject.billingProfileId.get)
```

Comment:
Nit: I think there should be a 3rd project here with a null or different billing profile ID to ensure it's not being returned erroneously.

Reply:
OK, will do.